R/dettl.R

```diff
@@ -58,8 +59,14 @@ DataImport <- R6::R6Class(
 
     reload = function() {
       dettl_config <- read_config(self$path)
+      mode <- dettl_config$dettl$mode
```

Comment:
This seems like more work than needed - can you not just pass the mode directly though your check function and have it return a validated one, so that this becomes

```
private$mode <- check_valid_mode(private$mode)
```

(or pass in `private$mode %||% "append"`) - I don't think you want to allow NA in here and in any case, that test will do unwanted things with `mode = c(NA, 1)` and `mode = c(1, NA)`.  In orderly we have a utility `match_value` that you might copy across for this bit of drudgery

Reply:
Thanks - I have tidied that up.